### PR TITLE
fix: inline style parsing for "! important"

### PR DESCRIPTION
**Pull request by Bradels**
## Does this pull request introduce a breaking change?

-   😮‍💨 No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

-   🔬 Yes, it does include an observable change.

If DISABLE_STATIC_CONTENT_OPTIMIZATION=1 "! important" will now be parsed correctly.
This approach uses Regex to find the !important tag at the end of the string, it is generous in that it allows for abitrary amounts of  white space.
Happy to update to only look for the one scenario outlined in the original issue.

### CODE REVIEW

**Comment by nolanlawson** [comment]
Hi @Bradels thanks for the PR! This looks good to me overall, but would you mind adding a test to confirm that we parse `! important` correctly? Here is an example of how we currently test this kind of feature:

https://github.com/salesforce/lwc/blob/05743434d7d24e8d068c3487df7c40d4376cb281/packages/%40lwc/integration-karma/test/rendering/programmatic-stylesheets/index.spec.js#L25-L27

In this case, we would need a component containing the ` ! important` style and then to confirm that `! important` is overriding some other CSS property correctly.

**Comment by Bradels** [comment]
Hi @nolanlawson Thank you for the reply.
I will make those tests and update this PR.
CLA still in progress, waiting on reponse from employer.

**Comment by Bradels** [comment]
Hi @nolanlawson , tests are added to confirm CSS important declaration is being parsed correctly and that it is overriding an existing CSS property that otherwise would be applied.

P.S. I may have updated my PR in an incorrect way, as it includes a commit that is not mine. If needed I can attempt to fix this by recreating my PR.

File: .github/workflows/karma.yml [hunk 1]
```diff
@@ -58,8 +58,6 @@ jobs:
             - run: FORCE_NATIVE_SHADOW_MODE_FOR_TEST=1 yarn sauce:ci
             - run: DISABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE=1 yarn sauce:ci
             - run: DISABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE=1 DISABLE_SYNTHETIC=1 yarn sauce:ci
-            - run: API_VERSION=58 yarn sauce:ci
-            - run: API_VERSION=58 DISABLE_SYNTHETIC=1 yarn sauce:ci
```
Comment (nolanlawson) [inline .github/workflows/karma.yml:62]: Looks like you are getting some unrelated changes here? Could you squash-merge or something?

Comment (nolanlawson) [reply .github/workflows/karma.yml]: Nevermind, I gotcha: adc8d9e05

File: packages/@lwc/template-compiler/src/codegen/helpers.ts [hunk 2]
```diff
@@ -273,10 +273,10 @@ export function styleMapToStyleDeclsAST(styleMap: { [name: string]: string }): t
     const styles: Array<[string, string] | [string, string, boolean]> = Object.entries(
         styleMap
     ).map(([key, value]) => {
-        const important = value.endsWith('!important');
+        const importantRegex = /\s*!\s*important\s*$/;
```
Comment (nolanlawson) [inline packages/@lwc/template-compiler/src/codegen/helpers.ts:276]:
```suggestion
        const importantRegex = /\s*!\s*important\s*$/i;
```

I looked into it, and this is actually case-insensitive ([codepen](https://codepen.io/nolanlawson-the-selector/pen/GRaavKZ)).

Could we add some tests for `!IMPORTANT` as well please?

File: packages/@lwc/integration-karma/test/rendering/style-specificity-important/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import Component from 'x/component';
+
+describe('important styling and style override', () => {
+    it('should render !important styles correctly', async () => {
+        const elm = createElement('x-component', { is: Component });
+        document.body.appendChild(elm);
+
+        await Promise.resolve();
+
+        for (const div of elm.shadowRoot.querySelectorAll('.imporant')) {
```
Comment (nolanlawson) [inline packages/@lwc/integration-karma/test/rendering/style-specificity-important/index.spec.js:11]:
```suggestion
        for (const div of elm.shadowRoot.querySelectorAll('.important')) {
```

Type here. 🙂

**Comment by Bradels** [comment]
Hey @nolanlawson, thank you for taking the time to look into my PR. Really grateful.
Will add your suggestions and capture the case insensitive scenario.

**Review by nolanlawson — APPROVED** [review]
LGTM! Thanks a bunch. 🙂
